restapi/admin_users.go

```diff
@@ -402,3 +411,109 @@ func getUpdateUserResponse(params admin_api.UpdateUserInfoParams) (*models.User,
 	}
 	return userElem, nil
 }
+
+// addUsersListToGroups iterates over the user list & assigns the requested groups to each user.
+func addUsersListToGroups(ctx context.Context, client MinioAdmin, usersToUpdate []string, groupsToAssign []string) error {
+	parallelGroupsUpdate := func(user, groupToAssign string) chan error {
+		chProcess := make(chan error)
+
+		go func() error {
+			defer close(chProcess)
+
+			userToAddRemove := []string{user}
+
+			updateReturn := updateGroupMembers(ctx, client, groupToAssign, userToAddRemove, false)
+
+			chProcess <- updateReturn
+
+			return updateReturn
+		}()
+
```

Comment:
go func doesn't need to return error, also updateGroupMembers returns error so it is easier to understand what it returns
so we should keep calling it `err` and handle it accordingly.
```suggestion
		go func() {
			defer close(chProcess)

			userToAddRemove := []string{user}

			err := updateGroupMembers(ctx, client, groupToAssign, userToAddRemove, false)

			if err != nil {
			   chProcess <- err
			   return // (this might not be needed)
			}
		}()
```

Reply:
Refactored function